### PR TITLE
Developer Guide doc updates

### PR DESCRIPTION
**Pull request by chrisglein**
* Split "Code Style and Conventions" and "Source Code Structure" out of the Developer Guide into separate documents (unchanged). Linked to them from the top of the Developer Guide.
* Updated information on checks and pipelines
* Fix broken links in the documentation (where %20 was being used for deep links instead of hyphen)

### CODE REVIEW

File: docs/developer_guide.md [hunk 1]
```diff
@@ -119,6 +33,11 @@ an Administrator Powershell window:
 
 `.\build\Install-WindowsSdkISO.ps1 18312`
```
Comment (jevansaks) [inline docs/developer_guide.md]: Can you update this to 18323? I forgot to in my last change.

Comment (jevansaks) [reply docs/developer_guide.md]:
```suggestion
`.\build\Install-WindowsSdkISO.ps1 18323
```

File: docs/source_code_structure.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+This is where the repo documentation lives, including this document.
+
+Note that developer usage documentation can be found separately on docs.microsoft.com.
+
+#### /test
+
+Our test library and test app (the app that the test library interacts with 
+when executing the tests) are here.
+
+MUXControls.Test is a [MSTest](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.testtools.unittesting)
```
Comment (licanhua) [inline docs/source_code_structure.md]: This supports both MSTest and TAEF test framework.

File: docs/source_code_structure.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Note that developer usage documentation can be found separately on docs.microsoft.com.
+
+#### /test
+
+Our test library and test app (the app that the test library interacts with 
+when executing the tests) are here.
+
+MUXControls.Test is a [MSTest](https://docs.microsoft.com/dotnet/api/microsoft.visualstudio.testtools.unittesting) 
+DLL using MITALite that contains all of the test code for the various controls
```
Comment (licanhua) [inline docs/source_code_structure.md]: remove MITALite

**Comment by chrisglein** [comment]
* Split "Code Style and Conventions" and "Source Code Structure" out of the Developer Guide into separate documents (unchanged). Linked to them from the top of the Developer Guide.
* Updated information on checks and pipelines
* Fix broken links in the documentation (where %20 was being used for deep links instead of hyphen)

**Pull request merged**